src/app/feedback/feedback.service.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import { AngularFireDatabase } from 'angularfire2/database';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { FeedbackService } from './feedback.service';
+import { inject, TestBed } from '@angular/core/testing';
+
+const router = {
+  url: 'abc123'
+};
+
+const activatedRoute = {
+  url: new BehaviorSubject<string>('')
+};
+
+const messages = [
+  {
+    name: 'a',
+    href: 'def456',
+    comment: 'blah',
+    email: 'dan@dan.com'
+  },
+  {
+    name: 'b',
+    href: 'abc123',
+    comment: 'blah',
+    email: 'dan@dan.com'
+  },
+  {
+    name: 'c',
+    href: 'abc123',
+    comment: 'blah',
+    email: 'dan@dan.com'
+  }
+];
+
+let messageStream: any = new BehaviorSubject(messages);
+messageStream = Object.assign(messageStream, { push: jasmine.createSpy('push') });
+
+const fire = {
+  list: jasmine.createSpy('list').and.returnValue(messageStream)
+};
+
+fdescribe('FeedbackService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        FeedbackService,
+        {
+          provide: AngularFireDatabase,
+          useValue: fire
+        },
+        { provide: Router, useValue: router },
+        { provide: ActivatedRoute, useValue: activatedRoute }
+      ]
+    });
+  });
+
+  it('should exist', inject([FeedbackService], (service: FeedbackService) => {
+    expect(service).toBeTruthy();
+  }));
+
+  it('should initialize data source when created', inject([FeedbackService], (service: FeedbackService) => {
+    expect(fire.list).toHaveBeenCalled();
+  }));
+
+  it('should list messages filtered by the router url', inject([FeedbackService, ActivatedRoute, Router],
```

Comment:
This test has an async call, can you add `async`? Are you sure this test is working or just passing because the subscribe block is never called to fail the expectation?